var/spack/repos/builtin/packages/exasp2/package.py

```diff
@@ -0,0 +1,83 @@
+# Produced at the Los Alamos National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Exasp2(MakefilePackage):
+    """ExaSP2 is a reference implementation of typical linear algebra algorithms
+    and workloads for a quantum molecular dynamics (QMD) electronic structure
+    code. The algorithm is based on a recursive second-order Fermi-Operator
+    expansion method (SP2) and is tailored for density functional based
+    tight-binding calculations of material systems. The SP2 algorithm variants
+    are part of the Los Alamos Transferable Tight-binding for Energetics
+    (LATTE) code, based on a matrix expansion of the Fermi operator in a
+    recursive series of generalized matrix-matrix multiplications. It is
+    created and maintained by Co-Design Center for Particle Applications
+    (CoPA). The code is intended to serve as a vehicle for co-design by
+    allowing others to extend and/or reimplement as needed to test performance
+    of new architectures, programming models, etc."""
+
+    tags = ['proxy-app']
+
+    homepage = "https://github.com/ECP-copa/ExaSP2"
+
+    version('develop', git='https://github.com/ECP-copa/ExaSP2',
+            branch='master')
+
+    variant('mpi', default=True, description='Build With MPI Support')
+
+    depends_on('bml')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('mpi', when='+mpi')
+    depends_on('bml@develop+mpi', when='+mpi')
+
+    build_directory = 'src'
+
+    @property
+    def build_targets(self):
+        targets = []
+        if '+mpi' in self.spec:
+            targets.append('PARALLEL=MPI')
+            targets.append('MPICC={0}'.format(self.spec['mpi'].mpicc))
+            targets.append('MPI_LIB=-L{0} -lmpi'.format(self.spec['mpi'].prefix.lib))
+            targets.append('MPI_INCLUDE=-I{0}'.format(self.spec['mpi'].prefix.include))
+        else:
+            targets.append('PARALLEL=NONE')
+        targets.append('BLAS=OPENBLAS')
```

Comment:
that's clearly wrong. It can be anything (openblas, atlas, intel-mkl, etc). See http://spack.readthedocs.io/en/latest/packaging_guide.html#blas-lapack-and-scalapack-libraries 

Try to specify Blas/Lapack libs explicitly. We do NOT want the build system to pick up something that is not consistent with what we promise users, i.e. threaded Atlas instead of non-threaded, or similar in case of MKL.

Reply:
> Add a comment that "OPENBLAS" stands for generic blas in their buildsystem.

if that's the case, one should still try to provide libs explicitly.

Reply:
It is the case:
https://github.com/ECP-copa/ExaSP2/blob/master/src/Makefile.vanilla#L131
but a

Reply:
then, just make the build system think it's `openblas` and hack-in proper link line in `LDFLAGS += -lopenblas`.  I see no reason why it should not work...

Reply:
all you need is:

```
math_libs = spec['lapack'].libs + spec['blas'].libs
math_libs.ld_flags  // <--- put that intsead of -lopenblas
```